src/middleware.ts

```diff
@@ -0,0 +1,27 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+export const languages = ['en', 'ru'];
+
+export function middleware(request: NextRequest) {
+    if (
+        request.nextUrl.pathname.startsWith('/api') ||
+        request.nextUrl.pathname.startsWith('/_next') ||
+        request.nextUrl.pathname.startsWith('/favicon') ||
+        request.nextUrl.pathname === '/'
```

Comment:
1. Why not redirect root? You should actually. Also, it's better to disable next locale detection.
2. What happens if a route already includes a locale other than current?